File: SDK/EdBApi/EdBApi/EdBApi.cpp
#include "EdBApi.h"
#include "EdBApiCore.h"

int EdB_Login(const char* IP, int Port, const char* Version, const char* User, const char* Password, const char* TXPass, const char* Yyb)
{
	return EdBApiCore::GetInstance()->login(IP, Port, Version, User, Password, TXPass, Yyb);
}

const char*  EdB_QueryData(const char* User, int ClientID, int Category)
{
	return EdBApiCore::GetInstance()->queryData(User, ClientID, Category);
}

const char* EdB_SendOrder(int ClientID, int Fx, const char *User, const char *Gddm, const char* Gpdm, int Quantity, float Price)
{
	return EdBApiCore::GetInstance()->sendOrder(ClientID, Fx, User, Gddm, Gpdm, Quantity, Price);
}

const char* EdB_GetPrice(int ClientID, const char * Gpdm)
{
	return EdBApiCore::GetInstance()->getPrice(ClientID, Gpdm);
}

const char* EdB_CancelOrder(int ClientID, const char * User, const char * Bho, int Jys)
{
	return EdBApiCore::GetInstance()->cancelOrder(ClientID, User, Bho, Jys);
}

File: Demos/EdBDemo_Java/EdBDemo_Java/src/EdBDemo_Java/EdBApi.java
package EdBDemo_Java;

import com.sun.jna.Library;
import com.sun.jna.Native;

public class EdBApi {
	public interface CLibrary extends Library
    {
		int EdB_Login(String IP, int Port, String Version, String User, String Password, String TXPass, String Yyb);
		String EdB_QueryData(String User, int ClientID, int Category);
		String EdB_SendOrder(int ClientID, int Fx, String User, String Gddm, String Gpdm, int Quantity, float Price);
		String EdB_GetPrice(int ClientID, String Gpdm);
		String EdB_CancelOrder(int ClientID, String User, String Bho, int jYs);
    }
	
	private static CLibrary lib = (CLibrary)Native.loadLibrary("EdBApi.dll",CLibrary.class); 
	
	public static int EdB_Login(String IP, int Port, String Version, String User, String Password, String TXPass, String Yyb)
	{
		return lib.EdB_Login(IP, Port, Version, User, Password, TXPass, Yyb);
	}
	
	public static String EdB_QueryData(String User, int ClientID, int Category)
	{
		return lib.EdB_QueryData(User, ClientID, Category);
	}
	
	public static String EdB_SendOrder(int ClientID, int Fx, String User, String Gddm, String Gpdm, int Quantity, float Price)
	{
		return lib.EdB_SendOrder(ClientID, Fx, User, Gddm, Gpdm, Quantity, Price);
	}
	
	public static String EdB_GetPrice(int ClientID, String Gpdm)
	{
		return lib.EdB_GetPrice(ClientID, Gpdm);
	}
	
	public static String EdB_CancelOrder(int ClientID, String User, String Bho, int jYs)
	{
		return lib.EdB_CancelOrder(ClientID, User, Bho, jYs);
	}
	
}
